src/client/_clientwindow.py

```diff
@@ -790,7 +790,10 @@ def updateOptions(self):
 
         self.gamelogs = self.actionSaveGamelogs.isChecked()
         self.player_colors.coloredNicknames = self.actionColoredNicknames.isChecked()
-        self.friendsontop = self.actionFriendsOnTop.isChecked()
+        if self.friendsontop != self.actionFriendsOnTop.isChecked():
+            self.friendsontop = self.actionFriendsOnTop.isChecked()
+            self.me.relationsUpdated.emit(self.me._friends._relations)
```

Comment:
'relationsUpdated' should be called if player's friend/foe relation actually changes, so I don't like this bit. What we could do instead is call _sortChatters on every chat channel (by adding some invalidate_sort method to Chat and calling that).